bundles/aliyun/src/main/java/org/apache/gravitino/oss/fs/OSSCredentialProvider.java

```diff
@@ -0,0 +1,132 @@
+  private final Configuration configuration;
+
+  public OSSCredentialProvider(URI uri, Configuration conf) {
+    this.filesetIdentifier =
+        conf.get(GravitinoVirtualFileSystemConfiguration.GVFS_FILESET_IDENTIFIER);
+    GravitinoVirtualFileSystem gravitinoVirtualFileSystem = new GravitinoVirtualFileSystem();
+    this.client = gravitinoVirtualFileSystem.initializeClient(conf);
+    this.configuration = conf;
+  }
+
+  @Override
+  public void setCredentials(Credentials credentials) {}
+
+  @Override
+  public Credentials getCredentials() {
+    // If the credentials are null or about to expire, refresh the credentials.
+    if (basicCredentials == null || System.currentTimeMillis() > expirationTime - 5 * 60 * 1000) {
+      synchronized (this) {
+        refresh();
+      }
+    }
+
+    return basicCredentials;
+  }
+
+  private void refresh() {
+    String[] idents = filesetIdentifier.split("\\.");
+    String catalog = idents[1];
+
+    FilesetCatalog filesetCatalog = client.loadCatalog(catalog).asFilesetCatalog();
+
+    Fileset fileset = filesetCatalog.loadFileset(NameIdentifier.of(idents[2], idents[3]));
+    Credential[] credentials = fileset.supportsCredentials().getCredentials();
+    if (credentials.length == 0) {
+      LOGGER.warn("No credential found for fileset: {}, try to use static AKSK", filesetIdentifier);
+      expirationTime = Long.MAX_VALUE;
+      this.basicCredentials =
+          new DefaultCredentials(
+              configuration.get(Constants.ACCESS_KEY_ID),
+              configuration.get(Constants.ACCESS_KEY_SECRET));
+      return;
+    }
+
+    Credential credential = getCredential(credentials);
+    Map<String, String> credentialMap = credential.toProperties();
+
+    String accessKeyId = credentialMap.get(GRAVITINO_OSS_SESSION_ACCESS_KEY_ID);
+    String secretAccessKey = credentialMap.get(GRAVITINO_OSS_SESSION_SECRET_ACCESS_KEY);
+
+    if (OSSTokenCredential.OSS_TOKEN_CREDENTIAL_TYPE.equals(
+        credentialMap.get(Credential.CREDENTIAL_TYPE))) {
+      String sessionToken = credentialMap.get(GRAVITINO_OSS_TOKEN);
+      this.basicCredentials = new BasicCredentials(accessKeyId, secretAccessKey, sessionToken);
+    } else {
+      this.basicCredentials = new DefaultCredentials(accessKeyId, secretAccessKey);
+    }
+
+    this.expirationTime = credential.expireTimeInMs();
+    if (expirationTime <= 0) {
+      expirationTime = Long.MAX_VALUE;
+    }
+  }
+
+  /**
+   * Get the credential from the credential array. Using dynamic credential first, if not found,
+   * uses static credential.
+   *
+   * @param credentials The credential array.
+   * @return The credential.
+   */
+  private Credential getCredential(Credential[] credentials) {
+    for (Credential credential : credentials) {
+      if (OSSTokenCredential.OSS_TOKEN_CREDENTIAL_TYPE.equals(credential.credentialType())) {
+        return credential;
+      }
+    }
+
+    // Not found, use the first one.
+    return credentials[0];
```

Comment:
should throw exception?

Reply:
I have checked the size of credentials and the size of it will always greater then 0

Reply:
It maybe some other credentials  like GCS credential

Reply:
Do you mean I can get a GCS credential when I fetch credentials for an OSS fileset?

Reply:
fileset will support multiple locations, and according to the API, there may other credentials besides OSS